lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -1039,13 +1039,14 @@ mkTxHistory wid txs = flatTxHistory
   where
     -- | Make flat lists of entities from the result of 'mkTxHistory'.
     flatTxHistory
-        :: [(TxMeta, ([TxIn], [TxOut]), [TxWithdrawal])]
-        -> ([TxMeta], [TxIn], [TxOut], [TxWithdrawal])
+        :: [(TxMeta, ([TxIn], [TxOut]), Maybe TxMetadata, [TxWithdrawal])]
+        -> ([TxMeta], [TxIn], [TxOut], [TxMetadata], [TxWithdrawal])
     flatTxHistory entities =
-        ( map (\(a,_,_) -> a) entities
-        , concatMap (fst . (\(_,b,_) -> b)) entities
-        , concatMap (snd . (\(_,b,_) -> b)) entities
-        , concatMap (\(_,_,c) -> c) entities
+        ( map (\(a,_,_,_) -> a) entities
+        , concatMap (fst . (\(_,b,_,_) -> b)) entities
+        , concatMap (snd . (\(_,b,_,_) -> b)) entities
+        , concatMap (\(_,_,c,_) -> maybeToList c) entities
+        , concatMap (\(_,_,_,d) -> d) entities
         )
```

Comment:
My reason was just that sqlite tables don't cost much and it seemed like it would be confusing to have data in the `TxMeta` table which was actually a field of `Tx`.